src/state/routing/args.ts

```diff
@@ -6,17 +6,15 @@ import { QuoteConfig, QuoteType } from 'hooks/routing/types'
 import useIsWindowVisible from 'hooks/useIsWindowVisible'
 import { useAtomValue } from 'jotai/utils'
 import { useMemo } from 'react'
-import { swapRouterUrlAtom } from 'state/swap'
+import { swapEventHandlersAtom, swapRouterUrlAtom } from 'state/swap'
 
 import { GetQuoteArgs } from './types'
 import { currencyAddressForSwapQuote } from './utils'
 
-const NON_SERIALIZABLE_KEYS = ['provider']
+const NON_SERIALIZABLE_KEYS = ['provider', 'onQuote']
 
-/** Omits the non-serializable keys from GetQuoteArgs' cache key. */
-export function serializeGetQuoteArgs({ endpointName, queryArgs }: { endpointName: string; queryArgs: GetQuoteArgs }) {
-  // same as default serializeQueryArgs, but ignoring non-serializable keys.
-  return `${endpointName}(${JSON.stringify(queryArgs, (key, value) => {
+export function serializeGetQuoteArgs(args: GetQuoteArgs) {
```

Comment:
is this function specific to `onQuote`/`GetQuote` or can we just rename it to be more generic (might help disambiguate between it and the new function below)

Reply:
These are specific to `GetQuote`:
- `serializeGetQuoteArgs` serializes a `GetQuoteArgs` object
- `serializeGetQuoteQueryArgs` is plugged into `serializeQueryArgs` for the `GetQuote` query

I think they're both named appropriately, just unfortunately very similarly.